container/libcontainer/compatibility.go

```diff
@@ -159,8 +195,22 @@ func ReadConfig(dockerRoot, dockerRun, containerID string) (*configs.Config, err
 
 		var state libcontainer.State
 		err = json.Unmarshal(out, &state)
-		if err != nil {
-			return nil, err
+		if err != nil && strings.Contains(err.Error(), "json: cannot unmarshal") {
```

Comment:
Should we check the type of this is a `json.UnmarshalTypeError` instead of checking the string contents?